solr/core/src/java/org/apache/solr/cloud/api/collections/Assign.java

```diff
@@ -551,11 +555,29 @@ public AssignStrategyFactory(SolrCloudManager solrCloudManager) {
       this.solrCloudManager = solrCloudManager;
     }
 
+    @SuppressWarnings("unchecked")
     public AssignStrategy create(ClusterState clusterState, CloudConfig cloudConfig, DocCollection collection) throws IOException, InterruptedException {
       @SuppressWarnings({"unchecked", "rawtypes"})
       List<Map> ruleMaps = (List<Map>) collection.get("rule");
       @SuppressWarnings({"rawtypes"})
       List snitches = (List) collection.get(SNITCH);
+      Map<String, Object> props = solrCloudManager.getClusterStateProvider().getClusterProperties();
+      Map<String,Object> assignInfo = (Map<String, Object>) props.get("assign-strategy");
```

Comment:
Do you have code (and curl) examples of how the `assign-strategy` property or similar ones are stored in `clusterprops.json`?

As I'm switching my PR to not use `solr.xml` but instead use `clusterprops.json`, I not only need to retrieve the plugin class name if any, but also associated configuration values whose names/keys and types are unknown since they are plugin specific.

If I copy the code as written here, I assume an iteration over `assignInfo` would reveal all configured values, want to make sure this makes sense.

Reply:
We already have an endpoint called  `/api/cluster/plugin`  GET/POST

Please refer to [SOLR-14404](https://issues.apache.org/jira/browse/SOLR-14404) for details

it can look as follows

```
curl -X POST -H 'Content-type:application/json' --data-binary '
{
  "set-assign-strategy": {
  "class": "full.ClassName", 
  "key1" : "val1"
  }
}' http://localhost:8983/api/cluster/plugins
```